lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex

```diff
@@ -194,4 +203,33 @@ defmodule LambdaEthereumConsensus.ForkChoice do
       {:ok, new_store}
     end
   end
+
+  @spec recompute_head(Types.Store.t()) :: :ok
+  def recompute_head(store) do
+    {:ok, head_root} = Helpers.get_head(store)
+
+    head_block = Store.get_block!(store, head_root)
+    head_execution_hash = head_block.body.execution_payload.block_hash
+
+    finalized_checkpoint = store.finalized_checkpoint
+    finalized_block = Store.get_block!(store, store.finalized_checkpoint.root)
+    finalized_execution_hash = finalized_block.body.execution_payload.block_hash
+
+    # TODO: do someting with the result from the execution client
+    # TODO: compute safe block hash
+    ExecutionClient.notify_forkchoice_updated(
+      head_execution_hash,
+      finalized_execution_hash,
+      finalized_execution_hash
+    )
+
+    BeaconChain.update_fork_choice_cache(
```

Comment:
We can return a map or a tuple with this and it can be caught by handle_info instead of calling a cast.